common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -496,6 +499,51 @@ public static void copyMemory(long srcAddr, byte[] dst, int dstIndex, long lengt
         PlatformDependent0.copyMemory(null, srcAddr, dst, BYTE_ARRAY_BASE_OFFSET + dstIndex, length);
     }
 
+    public static ByteBuffer allocateDirectNoCleaner(int capacity) {
+        if (DIRECT_MEMORY_COUNTER != null) {
+            for (;;) {
+                long used = DIRECT_MEMORY_COUNTER.get();
+                long expect = used + capacity;
```

Comment:
consider using a new name for these variables. `expect` terminology is typically the "expected value when trying to compareAndSet", but in this content is it used as what we want to set the value to.

```
long usedMemory = DIRECT_MEMORY_COUNTER.get();
long newUsedMemory = usedMemory + capacity;
```